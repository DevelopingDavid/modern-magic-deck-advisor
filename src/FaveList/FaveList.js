import React from 'react';
import FaveListItem from '../FaveListItem/FaveListItem.js'
import '../styles/main.scss';

function FaveList(props) {
  const faveDeckNames = props.faveDecks.map(deck => deck.deckName);
  faveDeckNames.sort();
  return(
    <div>
      <h1 className="list--h1">My Favorite Decks:</h1>
      <ul className="list--ul">
        {
          faveDeckNames.map((deck, index) => {
            return (
            <FaveListItem
              deckName={deck}
              deckIndex={index}
              key={deck}
              faveDecks={props.faveDecks}
              saveArray={props.saveArray}/>
            )
          })
        }
      </ul>
    </div>
  )
}

export default FaveList;