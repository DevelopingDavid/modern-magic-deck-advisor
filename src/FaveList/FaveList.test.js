import React from 'react';
import FaveList from './FaveList';
import { shallow } from 'enzyme';

describe('FaveList', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <FaveList/>
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})