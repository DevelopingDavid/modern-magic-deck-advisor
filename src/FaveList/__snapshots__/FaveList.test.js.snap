// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaveList should match the snapshot 1`] = `
<div>
  <h1
    className="list--h1"
  >
    Saved Decks:
  </h1>
  <ul
    className="list--ul"
  >
    <FaveListItem
      deckIndex={0}
      deckName="Bant Spirits"
      faveDecks={
        Array [
          Object {
            "deckName": "Bant Spirits",
          },
          Object {
            "deckName": "Izzet Phoenix",
          },
        ]
      }
      key="Bant Spirits"
      saveArray={[MockFunction]}
    />
    <FaveListItem
      deckIndex={1}
      deckName="Izzet Phoenix"
      faveDecks={
        Array [
          Object {
            "deckName": "Bant Spirits",
          },
          Object {
            "deckName": "Izzet Phoenix",
          },
        ]
      }
      key="Izzet Phoenix"
      saveArray={[MockFunction]}
    />
  </ul>
</div>
`;
